packages/payment-processor/src/payment/batch-conversion-proxy.ts

```diff
@@ -18,47 +20,65 @@ import {
 import { IPreparedTransaction } from './prepared-transaction';
 import { EnrichedRequest, IConversionPaymentSettings } from './index';
 import { checkRequestAndGetPathAndCurrency } from './any-to-erc20-proxy';
-import { getBatchArgs } from './batch-proxy';
 import { checkErc20Allowance, encodeApproveAnyErc20 } from './erc20';
-import { BATCH_PAYMENT_NETWORK_ID } from '@requestnetwork/types/dist/payment-types';
+import { BATCH_PAYMENT_NETWORK_ID, RequestDetail } from '@requestnetwork/types/dist/payment-types';
 import { IState } from 'types/dist/extension-types';
-import { CurrencyInput, isERC20Currency, isISO4217Currency } from '@requestnetwork/currency';
+import {
+  CurrencyInput,
+  isERC20Currency,
+  isISO4217Currency,
+  CurrencyManager,
+} from '@requestnetwork/currency';
+
+const currencyManager = CurrencyManager.getDefault();
 
 /**
  * Processes a transaction to pay a batch of requests with an ERC20 currency
- * that is different from the request currency (eg. fiat)
+ * that can be different from the request currency (eg. fiat)
  * The payment is made through ERC20 or ERC20Conversion proxies
  * It can be used with a Multisig contract
  * @param enrichedRequests List of EnrichedRequests to pay
  * @param version The version of the batch conversion proxy
  * @param signerOrProvider The Web3 provider, or signer. Defaults to window.ethereum.
+ * @param skipFeeUSDLimit Setting the value to true skips the USD fee limit, and reduce gas consumption.
+ *                        It can be useful to set it to false if the total amount of the batch is important.
+ *                        Check the value of batchFeeAmountUSDLimit of the batch proxy deployed.
  * @param overrides Optionally, override default transaction values, like gas.
- * @dev We only implement batchRouter using two ERC20 functions:
+ * @dev We only implement batchPayments using two ERC20 functions:
  *      batchMultiERC20ConversionPayments, and batchMultiERC20Payments.
  */
 export async function payBatchConversionProxyRequest(
   enrichedRequests: EnrichedRequest[],
   version: string,
   signerOrProvider: providers.Provider | Signer = getProvider(),
+  skipFeeUSDLimit = true,
```

Comment:
That's a bold move... I think it should be set to false by default: the risk of paying more gas is less important than the risk of paying more fees.

Reply:
I made this choice because it should be false only when a batch worth is above 50K $, which doesn't happen that often yet.
But if you think it is too risky, I am fine to set it to false by default.